lib/web/fetch/index.js

```diff
@@ -931,18 +933,32 @@ function schemeFetch (fetchParams) {
       // 4. Return a response whose status message is `OK`,
       //    header list is « (`Content-Type`, mimeType) »,
       //    and body is dataURLStruct’s body as a body.
-      return Promise.resolve(makeResponse({
+      return makeResponse({
         statusText: 'OK',
         headersList: [
           ['content-type', { name: 'Content-Type', value: mimeType }]
         ],
         body: safelyExtractBody(dataURLStruct.body)[0]
-      }))
+      })
     }
     case 'file:': {
       // For now, unfortunate as it is, file URLs are left as an exercise for the reader.
       // When in doubt, return a network error.
-      return Promise.resolve(makeNetworkError('not implemented... yet...'))
+      const fileURL = requestCurrentURL(request)
+
+      if (!process.permission?.has('fs.read', fileURL.href)) {
+        return makeNetworkError(`Access to ${fileURL.href} is not permitted.`)
+      }
+
+      try {
+        const blob = await openAsBlob(fileURL)
```

Comment:
using openAsBlob is the simplest solution; no need to manage an fd or creating a web readablestream ourselves

Reply:
fantastic!